src/hooks/useNavigation.js

```diff
@@ -4,10 +4,18 @@ import { useSoftkey } from 'hooks'
 export const useNavigation = (origin, containerRef, axis, elementsSelector = '[data-selectable]') => {
   const [current, setCurrent] = useState({ type: null, index: 0, key: null })
 
-  const getAllElements = () => document.querySelectorAll(elementsSelector)
+  const getAllElements = () => {
+    if (containerRef.current === undefined) {
+      return document.querySelectorAll(elementsSelector)
+    }
+
+    const grandparentContainer = containerRef.current.parentNode.parentNode
+    return grandparentContainer.querySelectorAll(elementsSelector)
+  }
 
   const getSelectedElement = () => {
-    return document.querySelector('[nav-selected=true]')
+    const grandparentContainer = containerRef.current.parentNode.parentNode
+    return grandparentContainer.querySelector('[nav-selected=true]')
```

Comment:
great, it works now!

two questions and comments

1. why grandparent, not just parent?
1. This fix. Although we will have the problem again if the grandparent or parent component use `useNavigation`, this is not our case as the container usually set to child components. I think we can ignore this now as this seems not a simple fix.

Reply:
In the case of `getSelectedElement`, we can actully get a working behavior if we just `return containerRef.current.querySelector('[nav-selected=true]’)` directly instead. The only issue with that is a harmless `TypeError` if you click on (invisible) center softkey when input element is active with a ListView/RadioListView below:

<img width="770" alt="image" src="https://user-images.githubusercontent.com/4752599/79899115-54f9fd00-83da-11ea-9b21-eef28815696c.png">

Reply:
> Although we will have the problem again if the grandparent or parent component use useNavigation, this is not our case as the container usually set to child components. I think we can ignore this now as this seems not a simple fix

I think the [latest commit](https://github.com/wikimedia/wikipedia-kaios/pull/210/commits/95900ec7965529e1e5bc95faea82b7e003ba5ac0) addresses this too then, if I'm understanding you correctly